test/Constraints/generics.swift

```diff
@@ -648,7 +648,7 @@ let arr = [BottleLayout]()
 let layout = BottleLayout(count:1)
 let ix = arr.firstIndex(of:layout) // expected-error {{referencing instance method 'firstIndex(of:)' on 'Collection' requires that 'BottleLayout' conform to 'Equatable'}}
 
-let _: () -> UInt8 = { .init("a" as Unicode.Scalar) } // expected-error {{missing argument label 'ascii:' in call}}
+let _: () -> UInt8 = { .init("a" as Unicode.Scalar) } // expected-error {{initializer 'init(_:)' requires that 'Unicode.Scalar' conform to 'BinaryInteger'}}
```

Comment:
😢 This change is a significant QoL regression for this diagnostic; the previous error was actually useful to the user, whereas this one is...not.

This is coming to my attention, incidentally, because #33889 will also cause this particular message to change; it's going to be brittle because _where_ an initializer is implemented (default implementation versus on the concrete type) now seems to change the diagnostic message.

Reply:
I am working on fixing this!